### DISCUSSION TITLE
useBalance for tokens with different decimals coming in wrong

### DISCUSSION THREAD

**Discussion by dev-cz** [wevm/wagmi#737] [Q&A] [answered]
When fetching the balance of a token through a token balance, the formatting comes in standard eth. But if the token has less decimals, it returns the value wrong. For example if getting balance for USDC balance with 6 decimals, it returns it as 0.000060 rather than 60.00.

Currently I am working around this using the below method in my app.

```
const inputTokenBalance = useBalance({
  addressOrName: address,
  token: inputTokenData?.address,
})
```

```
const inputBalance = inputTokenBalance.data?.value.toString() / (10 ** inputTokenBalance.data?.decimals)
```
    
  It would be great if wagmi used the formatted balance in conjunction with the token address decimal.

**Comment by tmm** [comment, marked as answer]
Fix in https://github.com/wagmi-dev/wagmi/pull/739

**Reply by dev-cz** [reply]
Awesome. Thank you!
